pkg/redirectpolicy/manager.go

```diff
@@ -101,18 +119,48 @@ func (rpm *Manager) RegisterSvcCache(cache svcCache) {
 // AddRedirectPolicy parses the given local redirect policy config, and updates
 // internal state with the config fields.
 func (rpm *Manager) AddRedirectPolicy(config LRPConfig) (bool, error) {
-	rpm.warnOnce.Do(func() {
-		if probes.HaveProgramHelper(ebpf.CGroupSockAddr, asm.FnSkLookupTcp) != nil ||
-			probes.HaveProgramHelper(ebpf.CGroupSockAddr, asm.FnSkLookupUdp) != nil {
-			log.Warn("Without socket lookup kernel functionality, BPF " +
-				"datapath cannot prevent potential loop caused by local-redirect" +
-				"service translation. Needs kernel version >= 5.1")
-		}
-	})
-
 	rpm.mutex.Lock()
 	defer rpm.mutex.Unlock()
 
+	if config.skipRedirectFromBackend {
+		rpm.noNetnsCookieSupport = sync.OnceValue[bool](func() bool {
+			if _, err := netns.GetNetNSCookie(); errors.Is(err, unix.ENOPROTOOPT) {
+				return true
+			}
+			rpm.epManager.Subscribe(rpm)
+			if rpm.skipLBMap == nil {
+				var err error
+				rpm.skipLBMap, err = lbmap.NewSkipLBMap()
+				if err != nil {
+					log.Warnf("failed to init cilium_skip_lb maps: %v"+
+						"policies with skipRedirectFromBackend flag set not supported", err)
+				}
+			}
+
+			return false
+		})()
+		if rpm.noNetnsCookieSupport {
+			err := fmt.Errorf("policy with skipRedirectFromBackend flag set not applied" +
+				":SO_NETNS_COOKIE not supported. Needs kernel version >= 5.8")
+			log.WithFields(logrus.Fields{
+				logfields.LRPType:      config.lrpType,
+				logfields.K8sNamespace: config.id.Namespace,
+				logfields.LRPName:      config.id.Name,
+			}).Error(err)
+			return false, err
+		}
+		if rpm.skipLBMap == nil {
+			err := fmt.Errorf("policy with skipRedirectFromBackend flag set not applied:" +
+				"requires cilium_skip_lb maps")
+			log.WithFields(logrus.Fields{
+				logfields.LRPType:      config.lrpType,
+				logfields.K8sNamespace: config.id.Namespace,
+				logfields.LRPName:      config.id.Name,
+			}).Error(err)
+			return false, nil
```

Comment:
Does the error need to be returned here?
```suggestion
			return false, err
```